File: testdata/jennies/rawtypes/field_with_struct_with_defaults/PHPRawTypes/src/Defaults/DefaultsStructPartialComplexField.php
<?php

namespace Grafana\Foundation\Defaults;

class DefaultsStructPartialComplexField implements \JsonSerializable
{
    public string $uid;

    public int $intVal;

    /**
     * @param string|null $uid
     * @param int|null $intVal
     */
    public function __construct(?string $uid = null, ?int $intVal = null)
    {
        $this->uid = $uid ?: "";
        $this->intVal = $intVal ?: 0;
    }

    public function equals(mixed $other): bool
    {
        if (!($other instanceof self)) {
            return false;
        }
    
        if ($this->uid !== $other->uid) {
            return false;
        }
    
        if ($this->intVal !== $other->intVal) {
            return false;
        }
    
        return true;
    }

    /**
     * @param array<string, mixed> $inputData
     */
    public static function fromArray(array $inputData): self
    {
        /** @var array{uid?: string, intVal?: int} $inputData */
        $data = $inputData;
        return new self(
            uid: $data["uid"] ?? null,
            intVal: $data["intVal"] ?? null,
        );
    }

    /**
     * @return mixed
     */
    public function jsonSerialize(): mixed
    {
        $data = new \stdClass;
        $data->uid = $this->uid;
        $data->intVal = $this->intVal;
        return $data;
    }
}

File: testdata/jennies/rawtypes/field_with_struct_with_defaults/JavaRawTypes/defaults/DefaultsStructPartialComplexField.java
package defaults;

import java.util.Objects;

public class DefaultsStructPartialComplexField {
    public String uid;
    public Long intVal;
    public DefaultsStructPartialComplexField() {
        this.uid = "";
        this.intVal = 0L;
    }
    public DefaultsStructPartialComplexField(String uid,Long intVal) {
        this.uid = uid;
        this.intVal = intVal;
    }

    @Override
    public boolean equals(Object other) {
        if (this == other) return true;
        if (!(other instanceof DefaultsStructPartialComplexField)) return false;
        DefaultsStructPartialComplexField o = (DefaultsStructPartialComplexField) other;
        if (!Objects.equals(this.uid, o.uid)) return false;
        if (!Objects.equals(this.intVal, o.intVal)) return false;
        return true;
    }

    @Override
    public int hashCode() {
        return Objects.hash(this.uid, this.intVal);
    }
}
